src/erlcloud_ec2.erl

```diff
@@ -1441,25 +1507,59 @@ extract_block_device_mapping_status(Node) ->
     ].
 
-%%
-%%
--spec(describe_instance_status/1 :: (string()) -> proplist()).
-describe_instance_status(InstanceID) ->
+-type instance_id() :: string().
+-type ec2_param_list() :: [{string(),string()}].
+
+-define(SPOT_INSTANCE_STATUS_MR_MIN, 5).
+-define(SPOT_INSTANCE_STATUS_MR_MAX, 1000).
+
+-spec describe_instance_status(instance_id()) -> ok_error(proplist()).
+describe_instance_status(InstanceID)
+    when is_list(InstanceID) ->
     describe_instance_status([{"InstanceId", InstanceID}], [], default_config()).
 
--spec(describe_instance_status/2 :: (proplist(), filter_list()) -> proplist()).
-describe_instance_status(Params, Filter) ->
+-spec describe_instance_status(ec2_param_list(), filter_list()) -> ok_error(proplist());
+                              (ec2_max_result(), ec2_token()) -> ok_error(proplist(), ec2_token()).
+describe_instance_status(Params, Filter)
+    when is_list(Params), is_list(Filter) ->
     describe_instance_status(Params, Filter, default_config()).
 
--spec(describe_instance_status/3 :: (proplist(), filter_list(), aws_config()) -> proplist()).
-describe_instance_status(Params, Filter, Config) ->
+-spec describe_instance_status(ec2_param_list(), filter_list(), aws_config()) -> ok_error(proplist()).
+describe_instance_status(Params, Filter, Config)
+    when is_list(Params), is_list(Filter), is_record(Config, aws_config) ->
     AllParams = Params ++ list_to_ec2_filter(Filter),
     case ec2_query(Config, "DescribeInstanceStatus", AllParams, ?NEW_API_VERSION) of
         {ok, Doc} ->
-            Path = "/DescribeInstanceStatusResponse/instanceStatusSet/item",
-            {ok, [ extract_instance_status(Item) || Item <- xmerl_xpath:string(Path, Doc) ]};
-        {error, _} = Error ->
-            Error
+            Statuses =
+                extract_results("DescribeInstanceStatusResponse", "instanceStatusSet", fun extract_instance_status/1, Doc),
+            {ok, Statuses};
+        {error, _} = E -> E
+    end.
+
+-spec describe_instance_status(ec2_param_list(), filter_list(), ec2_max_result(), ec2_token())
+        -> ok_error(proplist(), ec2_token()).
+describe_instance_status(Params, Filter, MaxResults, NextToken)
+    when is_list(Params), is_list(Filter), is_integer(MaxResults), is_list(NextToken) orelse NextToken =:= undefined ->
+    describe_instance_status(Params, Filter, MaxResults, NextToken, default_config()).
+
+-spec describe_instance_status(ec2_param_list(), filter_list(), ec2_max_result(), ec2_token(), aws_config())
+        -> ok_error(proplist(), ec2_token()).
+describe_instance_status(Params, Filter, MaxResults, NextToken, Config)
+    when is_list(Params),is_list(Filter),
+         is_integer(MaxResults), MaxResults >= ?SPOT_INSTANCE_STATUS_MR_MIN, MaxResults =< ?SPOT_INSTANCE_STATUS_MR_MAX,
+         is_list(NextToken) orelse NextToken =:= undefined,
+         is_record(Config, aws_config) ->
+    AllParams = Params ++
+                erlcloud_aws:single_param(MaxResults, "MaxResults") ++
```

Comment:
Param_list seems to be fine with both MaxResults and NextToken everywhere, seems like API deals with MaxResult and MaxResult.N formats equaly well. But some other params fail one of the tests in erlcloud_ec2_tests if they are used in that format. I might just remove single_param function and turn all single params back into [{"MaxResult", 5}] format for now